native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2TargetHandler.java

```diff
@@ -161,7 +161,15 @@ void writeContent(ChannelHandlerContext ctx, HttpContent msg) throws Http2Except
                         .setSenderState(new SendingHeaders(Http2TargetHandler.this, this));
                     outboundMsgHolder.setFirstContentWritten(true);
                 }
-                http2MessageStateContext.getSenderState().writeOutboundRequestBody(ctx, msg, http2MessageStateContext);
+                if (msg instanceof Http2ResetContent) {
+                    if (http2MessageStateContext.getSenderState() instanceof SendingHeaders) {
+                        http2MessageStateContext.getSenderState().writeOutboundRequestHeaders(ctx, msg);
```

Comment:
If `msg instanceof Http2ResetContent` is true, then simply set the state to `SendingRstFrame` and call its `writeOutboundRequestBody()` by removing `else` keyword in #L169

Reply:
When we set the state in `Http2TargetHandler`, we need to pass a `Http2RequestWriter` and everytime it is a new object. `SendingRstFrame` gets the corresponding `streamId` from `Http2RequestWriter` and will get `0` as the `streamId` in this case.

Reply:
When we set the state in `Http2TargetHandler`, we need to pass a `Http2RequestWriter` and everytime it is a new object. `SendingRstFrame` gets the corresponding `streamId` from `Http2RequestWriter` and will get `0` as the `streamId` in this case.

Reply:
Added the changes in https://github.com/ballerina-platform/module-ballerina-http/pull/659/commits/287830336fb2b11641e3547b9ee16b050e076d95